EssentialFeed/EssentialFeedTests/Feed API/LoadFeedImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class LoadFeedImageCommentsFromRemoteUseCaseTests: XCTestCase {
+    func test_init_doesNotRequestFromURL() {
+        let client = HTTPClientSpy()
+        _ = makeSUT()
+        
+        XCTAssertTrue(client.requestedURLs.isEmpty)
+    }
+    
+    func test_load_requestsFromURL() {
+        let url = anyURL()
+        let (sut, client) = makeSUT()
+        
+        _ = sut.load(from: url) { _ in }
+        
+        XCTAssertEqual(client.requestedURLs, [url])
+    }
+    
+    func test_loadTwice_requestsDataFromURLTwice() {
+        let url = anyURL()
+        let (sut, client) = makeSUT(url: url)
+        
+        _ = sut.load(from: url) { _ in }
+        _ = sut.load(from: url) { _ in }
+        
+        XCTAssertEqual(client.requestedURLs, [url, url])
+    }
+    
+    func test_load_deliversConnectivityErrorOnClientError() {
+        let (sut, client) = makeSUT()
+        let clientError = anyNSError()
+
+        expect(sut, toCompleteWith: failure(.connectivity), when: {
+            client.complete(with: clientError)
+        })
+    }
+    
+    func test_loadFromURL_deliversInvalidDataErrorOnNon200HTTPResponse() {
+        let (sut, client) = makeSUT()
+        let samples = [199, 201, 300, 400, 500]
+        
+        samples.enumerated().forEach { index, code in
+            expect(sut, toCompleteWith: failure(.invalidData), when: {
+                client.complete(withStatusCode: code, data: anyData(), at: index)
+            })
+        }
+    }
+    
+    func test_loadFromURL_deliversInvalidDataErrorOn200HTTPResponseWithEmptyData() {
```

Comment:
Rename test to match requirements:

```suggestion
    func test_loadFromURL_deliversInvalidDataErrorOn2xxHTTPResponseWithEmptyData() {
```